regexp/go.mod

```diff
@@ -0,0 +1,11 @@
+module github.com/hashicorp/go-secure-stdlib/regexp
+
+go 1.23
```

Comment:
didn't the `weak` package come out in 1.24? I guess that means we'll have to upgrade the Go version in order to use this?

Reply:
Fixed in 199d4648724bac134862d050023d2f623ddef39e